recipes-openxt/xenclient/ovmf/files/ovmf-q35-xen-support.patch

```diff
@@ -0,0 +1,662 @@
+ INF  OvmfPkg/SmmAccess/SmmAccessPei.inf
+ !endif
+ 
++
++
+ ################################################################################
+ 
+ [FV.DXEFV]
+@@ -207,12 +209,23 @@ INF  PcAtChipsetPkg/8259InterruptControllerDxe/8259.inf
+ INF  UefiCpuPkg/CpuIo2Dxe/CpuIo2Dxe.inf
+ INF  UefiCpuPkg/CpuDxe/CpuDxe.inf
+ INF  PcAtChipsetPkg/8254TimerDxe/8254Timer.inf
+-!if $(USE_OLD_PCI_HOST) == TRUE
+-INF  OvmfPkg/PciHostBridgeDxe/PciHostBridgeDxe.inf
+-!else
+-INF  MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciHostBridgeDxe.inf
++#!if $(USE_OLD_PCI_HOST) == TRUE
++#INF  OvmfPkg/PciHostBridgeDxe/PciHostBridgeDxe.inf
++#!else
++#INF  MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciHostBridgeDxe.inf
++#!endif
++#
++INF  DuetPkg/PciRootBridgeNoEnumerationDxe/PciRootBridgeNoEnumeration.inf
++INF  DuetPkg/PciBusNoEnumerationDxe/PciBusNoEnumeration.inf
++
++!if $(TPM_ENABLED) == TRUE
++INF SecurityPkg/Tcg/TcgConfigDxe/TcgConfigDxe.inf
++INF SecurityPkg/Tcg/TcgDxe/TcgDxe.inf
++#INF RuleOverride = DRIVER_ACPITABLE SecurityPkg/Tcg/TcgSmm/TcgSmm.inf
+ !endif
+-INF  MdeModulePkg/Bus/Pci/PciBusDxe/PciBusDxe.inf
++
++#INF  MdeModulePkg/Bus/Pci/PciBusDxe/PciBusDxe.inf
++
+ INF  PcAtChipsetPkg/KbcResetDxe/Reset.inf
+ INF  MdeModulePkg/Universal/Metronome/Metronome.inf
+ INF  PcAtChipsetPkg/PcatRealTimeClockRuntimeDxe/PcatRealTimeClockRuntimeDxe.inf
+diff --git a/OvmfPkg/PlatformPei/Platform.c b/OvmfPkg/PlatformPei/Platform.c
+index 7d09412..6cdea84 100644
+--- a/OvmfPkg/PlatformPei/Platform.c
++++ b/OvmfPkg/PlatformPei/Platform.c
+@@ -242,6 +242,8 @@ MemMapInitialization (
+     // 0xFEE00000    LAPIC                          1 MB
+     //
+     PciSize = 0xFC000000 - PciBase;
++
++
+     AddIoMemoryBaseSizeHob (PciBase, PciSize);
+     PcdSet64 (PcdPciMmio32Base, PciBase);
+     PcdSet64 (PcdPciMmio32Size, PciSize);
+diff --git a/OvmfPkg/PlatformPei/Xen.c b/OvmfPkg/PlatformPei/Xen.c
+index 7fa9019..f91bcb1 100644
+--- a/OvmfPkg/PlatformPei/Xen.c
++++ b/OvmfPkg/PlatformPei/Xen.c
+@@ -225,7 +225,19 @@ InitializeXen (
+   //
+   AddReservedMemoryBaseSizeHob (0xFC000000, 0x1000000, FALSE);
+ 
+-  PcdSetBool (PcdPciDisableBusEnumeration, TRUE);
++  //PcdSetBool (PcdPciDisableBusEnumeration, TRUE);
++
++  PcdSet64 (PcdPciMmio32Base, 0x80000000);
++  PcdSet64 (PcdPciMmio32Size, 0x7c000000);
++
++# if 0
++
++    AddIoMemoryBaseSizeHob (0xFEC00000, SIZE_4KB);
++    AddIoMemoryBaseSizeHob (0xFED00000, SIZE_1KB);
++    AddIoMemoryBaseSizeHob (ICH9_ROOT_COMPLEX_BASE, SIZE_16KB);
++    AddIoMemoryBaseSizeHob (PcdGet32(PcdCpuLocalApicBaseAddress), SIZE_1MB);
++#endif
++ 
+ 
+   return EFI_SUCCESS;
+ }
+diff --git a/OvmfPkg/XenBusDxe/XenBusDxe.c b/OvmfPkg/XenBusDxe/XenBusDxe.c
+index 439b060..86c4181 100644
+--- a/OvmfPkg/XenBusDxe/XenBusDxe.c
++++ b/OvmfPkg/XenBusDxe/XenBusDxe.c
+@@ -320,6 +320,8 @@ XenBusDxeDriverBindingStart (
+   XENIO_PROTOCOL *XenIo;
+   EFI_DEVICE_PATH_PROTOCOL *DevicePath;
+ 
++  return EFI_UNSUPPORTED;

```

Comment:
instead of just shorting out this function, can you not go to the invocations of this function and implement a check whether to even call?